### PR TITLE
Remove unnecessary zero checks from JVP rules in lax_linalg

### PR DESCRIPTION
**Pull request by shoyer**
I'm pretty sure all of these are dead code that can no longer be triggered, e.g., as evidenced by their use of no longer existing `core.pack` and `ad.TangentTuple`.

### CODE REVIEW

**Review by hawkinsp — COMMENTED** [review]
Hmm. I'm not sure they are actually unreachable, just that no-one has reached them in a while and they seem untested.

I think the correct behavior is now to return regular tuples, as the last example here does.

**Comment by shoyer** [comment]
I don't fully understand the ad interpreter, but experimentally it seems that if all the tangents are zero the JVP rule never gets called:
```python
import jax
from jax import core
from jax import ad

test_p = core.Primitive('test')
test_p.multiple_results = True
test_p.def_impl(lambda *args: args)

def _jvp_rule(x, dx):
  print('inside jvp!', x, dx)
  return x, dx

ad.primitive_jvps[test_p] = _jvp_rule

print('No zeros')
print(jax.jvp(lambda x: test_p.bind(x), (1.0,), (1.0,)))

print('All zeros -- no argument')
print(jax.jvp(lambda x: test_p.bind(1.0), (1.0,), (1.0,)))

print('All zeros -- stop gradient')
print(jax.jvp(lambda x: test_p.bind(jax.lax.stop_gradient(x)), (1.0,), (1.0,)))

print('Some zeros')
print(jax.jvp(lambda x: test_p.bind(x, 1.0), (1.0,), (1.0,)))
```
Prints:
```
No zeros
inside jvp! (1.0,) (1.0,)
([1.0], [1.0])
All zeros -- no argument
((1.0,), (array(0., dtype=float32),))
All zeros -- stop gradient
((1.0,), (array(0., dtype=float32),))
Some zeros
inside jvp! (1.0, 1.0) (1.0, Zero)
([1.0, 1.0], [1.0, array(0., dtype=float32)])
```

File: jax/lax_linalg.py [hunk 1]
```diff
@@ -326,8 +320,6 @@ def triangular_solve_shape_rule(a, b, left_side=False, **unused_kwargs):
 
 def triangular_solve_jvp_rule_a(
     g_a, ans, a, b, left_side, lower, transpose_a, conjugate_a, unit_diagonal):
-  if g_a is ad_util.zero:
```
Comment (hawkinsp) [inline jax/lax_linalg.py]: Even if the AD interpreter does not invoke the jvp rule if all inputs are symbolic zeros, this function has two inputs, and it is possible that only one is a symbolic zero.

**Review by hawkinsp — COMMENTED** [review]
Except for  the triangular solve rule this looks good.

File: jax/lax_linalg.py (hunk 1 above)
Comment (hawkinsp) [reply jax/lax_linalg.py]: Actually, I take it back. This primitive uses `defjvp2` which handles that case.

**Review by hawkinsp — APPROVED** [review]

**Pull request merged**